src/com/intellij/idea/plugin/hybris/project/configurators/impl/RegularContentRootConfigurator.java

```diff
@@ -328,6 +330,10 @@ private static <P extends JpsElement> void addSourceFolderIfNotIgnored(
         @NotNull final List<File> dirsToIgnore
     ) {
         if (dirsToIgnore.stream().noneMatch(it -> FileUtil.isAncestor(it, testSrcDir, false))) {
+            final boolean ignoreEmpty = HybrisApplicationSettingsComponent.getInstance().getState().getIgnoreNonExistingSourceDirectories();
```

Comment:
@viktor-yengovatov , what do you think if we can refactor this part slightly and pass state of the `HybrisApplicationSettingsComponent` to this class -> methods from the `ImportProjectProgressModalWindow`
by that we will get appSettings only once during import of all java modules?

variable can be declared in the `public synchronized void run(@NotNull final ProgressIndicator indicator) {` of the `ImportProjectProgressModalWindow` and then passed to `createJavaModule` then to `configuratorFactory.getContentRootConfigurator(moduleDescriptor).configure` and so on

Reply:
Hello @mlytvyn 
I have created separate pull request for the refactoring: https://github.com/epam/sap-commerce-intellij-idea-plugin/pull/512 , #512 .
I hope it fits.

Reply:
sure, just take into account different target branch, I'll also change target branch for your PR #512